### ISSUE TITLE
document.write() no longer supported on mobile chrome

### ISSUE THREAD

**Issue by ahayes** [GCRC/nunaliit#725]
See this older update notice from Google. https://developers.google.com/web/updates/2016/08/removing-document-write

Note that we appear to use document.write() in our Nunaliit JS and this may cause issues on mobile and was discovered by one of our app developer candidates as being a problem for the webviews in the nunaliit_android project.

**Comment by ahayes** [comment]
Notes from dev:

The project consists of a native Java part describing communication and data transfer from CouchDB and a hybrid Web-based part of the Cordova framework. The web part is responsible for displaying the JavaScript code generated by the Nulaliit framework. The web part uses configuration scripts with the “debug” suffix and without it the content seems to be a minified javascript. For example, nunaliit2-couch.js and nunaliit2-couch-debug.js. Looks like the debug version of the script prepares all declared scripts inside it and injects the result into the web form body with the help of the ‘document.write()’ command.

When debugging the Nunaliit-Android project in the Chrome Developers Tools we noticed the warnings about scripts being blocking. Those scripts were calling ‘document.write()’:

Looks like it was the reason for a WebView not being able to display content on Android. According to Google resources for Web Developers ‘document.write()’ usage impacts the performance of the page rendering and therefore Google blocked ‘document.write()’ in Chromium engine, which in fact is a core for Android WebView.

**Comment by jpfiset** [comment]
I have combined all the components for nunaliit2 and nunaliit2-couch into one compressed file: nunaliit2.js  Also, the build now creates two version of that file: a debug version (nunaliit2-debug.js) and a debug inline version (nunaliit2-debug-il.js). The latter might be useful for mobile apps.

The files nunaliit2-couch.js and nunaliit2-couch-debug.js are still produced but are empty. This is to not generate errors on legacy sites that still reference them.

Finally, the mobile libraries are still generated, but the debug version has been changed to inline. During Android app development, this is the file that should be loaded.
